File: FrontEnd (C#)/SoftProgBusiness/GestMaterial/BOI/ArticuloBOImpl.cs
using SoftProgBusiness.GestMaterial.BO;
using SoftProgModel.GestMaterial;
using SoftProgPersistance.GestMaterial.DAO;
using SoftProgPersistance.GestMaterial.Impl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.GestMaterial.BOI
{
    public class ArticuloBOImpl : IArticuloBO
    {
        private ArticuloDAO daoArticulo;
        public ArticuloBOImpl()
        {
            daoArticulo = new ArticuloImpl();
        }
        public int eliminar(int idObjeto)
        {
            return daoArticulo.eliminar(idObjeto);
        }

        public int insertar(Articulo objeto)
        {
            return daoArticulo.insertar(objeto);
        }

        public BindingList<Articulo> listarTodos()
        {
            return daoArticulo.listarTodos();
        }

        public int modificar(Articulo objeto)
        {
            return daoArticulo.modificar(objeto);   
        }

        public Articulo obtenerPorId(int idObjeto)
        {
            return daoArticulo.obtenerPorId(idObjeto);  
        }

        public void validar(Articulo objeto)
        {
            throw new NotImplementedException();
        }
    }
}

File: BackEnd (Java)/BibliotecaBusiness/src/main/java/biblioteca/gestionMaterial/boImpl/ArticuloBOImpl.java

package biblioteca.gestionMaterial.boImpl;

import biblioteca.gestionMaterial.bo.ArticuloBO;
import biblioteca.gestionMaterial.dao.ArticuloDAO;
import biblioteca.gestionMaterial.mysql.Articulolmpl;
import java.util.ArrayList;
import pe.edu.pucp.utilsarmy.gestion_de_material.model.Articulo;

public class ArticuloBOImpl implements ArticuloBO{
    private final ArticuloDAO articulodao;

    public ArticuloBOImpl() {
        articulodao=new Articulolmpl();
    }
    
    @Override
    public int insertar(Articulo objeto) throws Exception {
        return articulodao.insertar(objeto);
    }

    @Override
    public int modificar(Articulo objeto) throws Exception {
        return articulodao.modificar(objeto);
    }

    @Override
    public int eliminar(int idObjeto) throws Exception {
        return articulodao.eliminar(idObjeto);
    }

    @Override
    public Articulo obtenerPorId(int idObjeto) throws Exception {
        return articulodao.obtenerPorId(idObjeto);
    }

    @Override
    public ArrayList<Articulo> listarTodos() throws Exception {
        return articulodao.listarTodos();
    }

    @Override
    public void validar(Articulo objeto) throws Exception {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }
    
}
